File: frontend/src/api/aiSettings.ts
import { apiClient } from './client';

export interface AISettings {
  id: string;
  user_id: string;
  confidence_threshold: number;
  strategy_type: 'conservative' | 'balanced' | 'aggressive';
  trade_amount: number;  // 🔥 ADD THIS
  stop_loss_percent: number;
  take_profit_percent: number;
  max_daily_loss: number;
  max_drawdown: number;
  max_positions: number;
  risk_per_trade: number;
  position_size_multiplier: number;
  symbols: string[];
  auto_trade_enabled: boolean;
  max_trades_per_day: number;
  total_trades: number;
  winning_trades: number;
  losing_trades: number;
  total_pnl: number;
  best_trade: number;
  worst_trade: number;
  created_at: string;
  updated_at: string;
  win_rate?: number;
  strategy_label?: string;
}

export interface StrategyPreset {
  type: string;
  label: string;
  description: string;
  icon: string;
  risk_level: string;
  settings: {
    confidence_threshold: number;
    stop_loss_percent: number;
    take_profit_percent: number;
    position_size_multiplier: number;
    max_positions: number;
    trade_amount: number;
  };
}

export interface PerformanceStats {
  total_trades: number;
  winning_trades: number;
  losing_trades: number;
  win_rate: number;
  total_pnl: number;
  best_trade: number;
  worst_trade: number;
}

export const aiSettingsApi = {
  getSettings: () => apiClient.get<AISettings>('/ai-settings'),
  updateSettings: (data: Partial<AISettings>) => apiClient.put<AISettings>('/ai-settings', data),
  getPresets: () => apiClient.get<StrategyPreset[]>('/ai-settings/presets'),
  applyPreset: (presetType: string) =>
    apiClient.post<{ message: string; settings: AISettings }>(
      `/ai-settings/apply-preset/${presetType}`
    ),
  getSymbols: () =>
    apiClient.get<{ symbol: string; name: string; icon: string }[]>('/ai-settings/symbols'),
  getPerformance: () =>
    apiClient.get<PerformanceStats>('/ai-settings/performance-stats'),
  toggleAutoTrade: (enabled: boolean) =>
    apiClient.post<{ auto_trade_enabled: boolean; message: string }>(
      '/ai-settings/toggle-auto-trade',
      null,
      { params: { enabled } }
    ),
};

File: backend/app/schemas/ai_settings.py
"""
AI Settings Pydantic Schemas - Professional Trading Configuration
"""

from pydantic import BaseModel, Field
from typing import List, Optional
from datetime import datetime

# ============================================
# STRATEGY INFO SCHEMA
# ============================================
class StrategyInfo(BaseModel):
    type: str
    label: str
    description: str
    icon: str
    risk_level: str
    settings: dict

# ============================================
# STRATEGY PRESET SCHEMA
# ============================================
class StrategyPreset(BaseModel):
    type: str
    label: str
    description: str
    icon: str
    risk_level: str
    settings: dict

# ============================================
# AI SETTINGS BASE SCHEMA
# ============================================
class AISettingsBase(BaseModel):
    confidence_threshold: float = Field(70.0, ge=50, le=90)
    strategy_type: str = Field('balanced')
    
    # 🔥 PER-TRADE AMOUNT
    trade_amount: float = Field(25.0, ge=1, le=10000)
    
    stop_loss_percent: float = Field(2.0, ge=0.5, le=10)
    take_profit_percent: float = Field(4.0, ge=1, le=20)
    max_daily_loss: float = Field(5.0, ge=0.5, le=30)
    max_drawdown: float = Field(15.0, ge=5, le=50)
    max_positions: int = Field(5, ge=1, le=20)
    risk_per_trade: float = Field(2.0, ge=0.5, le=5)
    position_size_multiplier: float = Field(1.0, ge=0.1, le=3.0)
    symbols: List[str] = ['BTC/USDT', 'ETH/USDT', 'SOL/USDT', 'BNB/USDT']
    auto_trade_enabled: bool = False
    max_trades_per_day: int = Field(10, ge=1, le=50)

class AISettingsCreate(AISettingsBase):
    pass

class AISettingsUpdate(BaseModel):
    confidence_threshold: Optional[float] = Field(None, ge=50, le=90)
    strategy_type: Optional[str] = None
    trade_amount: Optional[float] = Field(None, ge=1, le=10000)
    stop_loss_percent: Optional[float] = Field(None, ge=0.5, le=10)
    take_profit_percent: Optional[float] = Field(None, ge=1, le=20)
    max_daily_loss: Optional[float] = Field(None, ge=0.5, le=30)
    max_drawdown: Optional[float] = Field(None, ge=5, le=50)
    max_positions: Optional[int] = Field(None, ge=1, le=20)
    risk_per_trade: Optional[float] = Field(None, ge=0.5, le=5)
    position_size_multiplier: Optional[float] = Field(None, ge=0.1, le=3.0)
    symbols: Optional[List[str]] = None
    auto_trade_enabled: Optional[bool] = None
    max_trades_per_day: Optional[int] = Field(None, ge=1, le=50)

class AISettingsResponse(AISettingsBase):
    id: str
    user_id: str
    total_trades: int = 0
    winning_trades: int = 0
    losing_trades: int = 0
    total_pnl: float = 0.0
    best_trade: float = 0.0
    worst_trade: float = 0.0
    created_at: datetime
    updated_at: Optional[datetime]
    
    class Config:
        from_attributes = True